pkg/controller/manager/nad/controller.go

```diff
@@ -125,6 +135,43 @@ func (h Handler) OnRemove(key string, nad *cniv1.NetworkAttachmentDefinition) (*
 	return nad, nil
 }
 
+func (h Handler) ensureLabels(nad *cniv1.NetworkAttachmentDefinition) error {
+	if nad.Labels != nil && nad.Labels[utils.KeyNetworkType] != "" && nad.Labels[utils.KeyClusterNetworkLabel] != "" {
+		return nil
+	}
+
+	labels := nad.Labels
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+
+	netconf := &utils.NetConf{}
+	if err := json.Unmarshal([]byte(nad.Spec.Config), netconf); err != nil {
+		return fmt.Errorf("unmarshal config of nad %s/%s failed, error: %w", nad.Namespace, nad.Name, err)
+	}
+
+	lenOfBrName := len(netconf.BrName)
+	if lenOfBrName <= len(iface.BridgeSuffix) {
+		return fmt.Errorf("the length of brName must be more than %d", len(iface.BridgeSuffix))
+	}
```

Comment:
prefer not to check the bridgeSuffix via controller loop, if needed, move it to the webhook validator.